kbase-extension/static/kbase/js/common/jobManager.js

```diff
@@ -374,34 +377,94 @@ define([
                 this.removeChannelListeners(channelType, channelId);
             });
         }
+    }
 
-        /**
-         * Update the model with the supplied jobState objects
-         * @param {array} jobArray list of jobState objects to update the model with
-         */
-        updateModel(jobArray) {
-            const jobIndex = this.model.getItem('exec.jobs');
-            const batchId = this.model.getItem('exec.jobState.job_id');
-            let batchJob;
-            jobArray.forEach((jobState) => {
-                // update the job object
-                jobIndex.byId[jobState.job_id] = jobState;
-                if (jobState.job_id === batchId) {
-                    batchJob = jobState;
+    const SingleJobMixin = (Base) =>
+        class extends Base {
+            constructor(config) {
+                // run the constructor for the base class
+                super(config);
+                this.pollInterval = config.pollInterval || DEFAULT_STATUS_REQUEST_INTERVAL;
+            }
+
+            init() {
+                const job = this.model.getItem('exec.jobState');
+                if (!job) {
+                    throw new Error('Cannot init without a job');
                 }
-            });
-            this.model.setItem('exec.jobs', jobIndex);
-            // check whether the batch parent needs updating
-            if (batchJob) {
-                this.model.setItem('exec.jobState', batchJob);
+
+                const self = this;
+                ['ERROR', 'INFO', 'LOGS', 'STATUS'].forEach((msgType) => {
+                    this.addListener(jcm.MESSAGE_TYPE[msgType], jcm.CHANNEL.JOB, job.job_id);
+                });
+
+                if (!this.looper) {
+                    this.looper = new Looper({ pollInterval: this.pollInterval });
+                }
+
+                this.addEventHandler(jcm.MESSAGE_TYPE.STATUS, {
+                    // default job status handler
+                    [`__default_${jcm.MESSAGE_TYPE.STATUS}`]: self.handleJobStatus.bind(self),
+
+                    // set up a delayed request for job status, executed after this.pollInterval ms
+                    jobStatusLooper: () => {
+                        self.looper.scheduleRequest(self.requestJobStatus.bind(self));
+                    },
```

Comment:
a couple of default handlers -- the first processes and saves job status messages to the model, and the second sets up a looping status update request